p2p/host/basic/basic_host.go

```diff
@@ -104,6 +104,7 @@ func New(net inet.Network, opts ...interface{}) *BasicHost {
 
 // newConnHandler is the remote-opened conn handler for inet.Network
 func (h *BasicHost) newConnHandler(c inet.Conn) {
+	h.Peerstore().SetProtocols(c.RemotePeer())
```

Comment:
we have issues around remembering protocols across connection resets. For now we're just clearing them